src/odb/include/odb/dbStream.h

```diff
@@ -398,9 +413,31 @@ class dbIStream
     return *this;
   }
 
+  template <uint32_t I = 0, typename... Ts>
+  dbIStream& operator>>(std::variant<Ts...>& v)
+  {
+    uint32_t index = 0;
+    *this >> index;
+    return variantHelper(index, v);
+  }
+
   double lefarea(int value) { return ((double) value * _lef_area_factor); }
 
   double lefdist(int value) { return ((double) value * _lef_dist_factor); }
+
+ private:
+  template <uint32_t I = 0, typename... Ts>
+  dbIStream& variantHelper(uint32_t index, std::variant<Ts...>& v)
+  {
+    if constexpr (I == sizeof...(Ts)) {
+      return *this;
+    } else {
+      if (I == index) {
+        *this >> std::get<I>(v);
+      }
+      return ((*this).operator>><I + 1>(v));
```

Comment:
What's the template recursion limit?

Reply:
Line 432, when I is equal to the number of elements in the variant, then we just return dbIStream, else we do a recursion to the next element.

Reply:
There's a C++ compiler limit to the recursion depth though, at least I thought there was. I also wouldn't be entirely surprised if its lower inside Google than default configurations of LLVM.

Reply:
looks like it's around 500 https://stackoverflow.com/questions/12277732/template-metaprogramming-recursion-up-limits

Reply:
Looking inside Google it appears this limit is 40, which is a lot lower. Can we include a compile error in the template at 39 to let the user they need to do something else.

Reply:
Yeah, seems to be controlled by -ftemplate-depth. I looked and I could not find one source of true, but values seems to vary from 256 to 1024.

I don't think there is a use case for variants bigger than a few elements, do you want me to put a limit, let's say 16 so future usage of big variants fail to compile? I think it is better than introducing possible sources of incompatibility for future compilations.

Reply:
Anybody using a variant with 40+ members will have to use a real data structure instead....

Reply:
Yeah 16 feels like the right number